ec_sense.py

```diff
@@ -0,0 +1,154 @@
+"""
+Organization: Professorship of Environmental Sensing and Modelling, TU Munich
+Date: 08.03.2022
+Author: Daniel Kühbacher
+
+Description: This class can be used to read out EC gas modules.
+"""
+
+import serial
+from time import sleep
+
+class ec_sensor:
+    
+    types = {'0x21': 'NO2', '0x23': 'O3', '0x19': 'CO'}
+    units = {'0x2': 'ppm', '0x4': 'ppb', '0x8': '%'}
+
+    def __init__(self, port):
+        """
+            ### Constructor ###
+            Establishes connection to sensor and checks the sensor type, unit and measurment range
+        """
+        
+        try:
+            # connect to sensor
+            self.ser = serial.Serial(port,
+                                baudrate = 9600,
+                                parity = serial.PARITY_NONE,
+                                stopbits = serial.STOPBITS_ONE,
+                                bytesize = serial.EIGHTBITS,
+                                timeout = 1)
```

Comment:
Retry: Try n times until failing

Reply:
https://github.com/jd/tenacity

Reply:
nutze hier vorerst eine for loop. war schneller implementiert da ich bei tenacity permanent fehler bekomme